### DISCUSSION TITLE
WebView2 do not support 【crypto.randomuuID()】?

### DISCUSSION THREAD

**Discussion by yinhaide** [MicrosoftEdge/WebView2Feedback#4238] [Q&A] [answered]
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/7472758/5062857d-30d2-4ffe-894c-92808e456feb)

I am loading a local HTML file using WebView2 and trying to use crypto.randomUUID(), but it is not supported. How can I use this API?
Microsoft.Web.WebView2: 1.0.2088.41
WebView2.CoreWebView2.Environment.BrowserVersionString: 120.0.2210.61
Microsoft Edge: 120.0.2210.61
IDE：Visual Studio
APP：VSIX Extension

**Comment by yinhaide** [comment, marked as answer]
use https is ok~

**Reply by victorhuangwq** [reply]
crypto.randomuuID is only supported in the HTTPS secure context: https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID, as such you it works in https, but not in http.
